pkg/svcat/service-catalog/binding_test.go

```diff
@@ -0,0 +1,213 @@
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Binding", func() {
+	var (
+		sdk          *SDK
+		svcCatClient *fake.Clientset
+		sb           *v1beta1.ServiceBinding
+		sb2          *v1beta1.ServiceBinding
+	)
+
+	BeforeEach(func() {
+		sb = &v1beta1.ServiceBinding{ObjectMeta: metav1.ObjectMeta{Name: "foobar", Namespace: "foobar_namespace"}}
+		sb2 = &v1beta1.ServiceBinding{ObjectMeta: metav1.ObjectMeta{Name: "barbaz", Namespace: "foobar_namespace"}}
+		svcCatClient = fake.NewSimpleClientset(sb, sb2)
+		sdk = &SDK{
+			ServiceCatalogClient: svcCatClient,
+		}
+	})
+
+	Describe("RetrieveBinding", func() {
+		It("Calls the generated v1beta1 Get method with the passed in binding and namespace", func() {
+			binding, err := sdk.RetrieveBinding(sb.Namespace, sb.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(binding.Name).To(Equal(sb.Name))
+
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "servicebindings")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(sb.Name))
+			Expect(actions[0].(testing.GetActionImpl).Namespace).To(Equal(sb.Namespace))
+		})
+		It("Bubbles up errors", func() {
+			fakeName := "not_a_real_binding"
+
+			_, err := sdk.RetrieveBinding(sb.Namespace, fakeName)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("not found"))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "servicebindings")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(fakeName))
+			Expect(actions[0].(testing.GetActionImpl).Namespace).To(Equal(sb.Namespace))
+		})
+	})
+
+	Describe("RetrieveBindings", func() {
+		It("Calls the generated v1beta1 List method with the specified namespace", func() {
+			bindings, err := sdk.RetrieveBindings(sb.Namespace)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bindings.Items).Should(ConsistOf(*sb, *sb2))
+			Expect(svcCatClient.Actions()[0].Matches("list", "servicebindings")).To(BeTrue())
+		})
+		It("Bubbles up errors", func() {
+			badClient := &fake.Clientset{}
+			errorMessage := "error retrieving list"
+			badClient.AddReactor("list", "servicebindings", func(action testing.Action) (bool, runtime.Object, error) {
+				return true, nil, fmt.Errorf(errorMessage)
+			})
+			sdk.ServiceCatalogClient = badClient
+
+			bindings, err := sdk.RetrieveBindings(sb.Namespace)
+
+			Expect(bindings).To(BeNil())
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring(errorMessage))
+			Expect(badClient.Actions()[0].Matches("list", "servicebindings")).To(BeTrue())
+		})
+	})
+
+	Describe("RetrieveBindingsByInstance", func() {
+		It("Calls the generated v1beta1 List method on the provided instance's namespace", func() {
+			si := &v1beta1.ServiceInstance{ObjectMeta: metav1.ObjectMeta{Name: "apple_instance", Namespace: sb.Namespace}}
+			sb.Spec.ServiceInstanceRef.Name = si.Name
+			svcCatClient = fake.NewSimpleClientset(sb, sb2)
```

Comment:
Are lines 104-107 still needed? I would have expected that the `BeforEach` is sufficient.

Reply:
I have to recreate the client so the sb is connected to the service instance - they're passed in by value so just setting sb.Spec.ServiceInstanceRef.Name doesn't work.

Reply:
Thanks, I appreciate the explanations! 👍